support/build/Makefile.rules

```diff
@@ -459,6 +459,26 @@ define fetch2 =
 $(call fetchas2,$(1),$(2),$(3),$(notdir $(2)))
 endef
 
+# Downloads a repo from remote
+# This command can not clone a specific commit from a
+# specific branch of a repo
+# clone $libname $giturl $branch/tag $dir_with_patches(optional)
+define clone =
+$(BUILD_DIR)/$(1)/.origin:
+	$(call verbose_cmd,RM,$(1),$(RM) -r $(BUILD_DIR)/$(1)/origin)
+	$(call verbose_cmd,GIT,$(1),$(GIT) clone -q --progress --depth 1 --branch $(3) --no-tags $(2) $(BUILD_DIR)/$(1)/origin/ && $(TOUCH) $(BUILD_DIR)/$(1)/.origin)
```

Comment:
Would it make sense to also clone the submodules? `--recurse-submodules`

Reply:
This is an interesting idea. Is the parameter `--recurse-submodules` harmful when no submodiules exist (e.g., in terms of exit code of `git`) . If not we could add this parameter.

Reply:
No. It does not. Will add it to the command. Thanks, @vladandrew.

Reply:
Added.